stdlib/public/runtime/MetadataLookup.cpp

```diff
@@ -748,3 +791,39 @@ swift_getTypeByMangledName(const char *typeNameStart, size_t typeNameLength,
 
   return Demangle::decodeMangledType(builder, node);
 }
+
+std::pair<std::string, FieldType> swift_getTypeFieldAt(const Metadata *type,
+                                                       unsigned index) {
+  auto &fields = TypeFields.get();
+  auto mangledName = swift_getTypeName(type, true);
+
+  auto getFieldAt = [&](const FieldDescriptor &descriptor,
+                        unsigned i) -> std::pair<std::string, FieldType> {
+    auto &field = descriptor.getFields()[i];
+    auto name = field.getFieldName(0);
+    auto type = field.getMangledTypeName(0);
+
+    auto *typeMetadata = swift_getTypeByMangledName(type.data(), type.length(),
+                                                    0, nullptr, nullptr);
```

Comment:
Until we're able to provide the substitutions to `swift_getTypeByMangledName`, this won't work for generic types. There's no good way to get all of those substitutions correctly now (the information isn't there in the nominal type descriptors), but you should be able to handle non-nested generics with no requirements by using `TargetMetadata::getGenericArgs()`. That would at least be enough to see basic substitutions working.

Fully supporting substitutions here will require @jckarter  to land https://github.com/apple/swift/pull/13468

Reply:
Good to know, thanks, @DougGregor ! In a state I have it right now, would it be able to find non-generic types? Also, all of the methods in `Field{Descriptor, Record}` require `Offset` as an argument, I looked at how `swift-reflection-dump` gets it, but I don't think it's possible to do the same in `ImageInspection{ELF, MachO}` because we there is no `SectionRef`, is there some other way to get it without using object file interface?

Reply:
Talked to @jckarter offline and it looks like I don't need to worry about `offset` I mentioned in my previous comment. So the only thing which remains is to get some information from given `type` to supply generic env to the `swift_getTypeByMangledName` call and wait for other changes to be landed.